src/components/Notifications/index.js

```diff
@@ -0,0 +1,150 @@
+} from '#request';
+
+import _ts from '#ts';
+
+import ProjectJoinRequestItem from './items/ProjectJoinRequest';
+import ProjectJoinRequestAbortItem from './items/ProjectJoinRequestAbort';
+import ProjectJoinResponseItem from './items/ProjectJoinResponse';
+
+import styles from './styles.scss';
+
+const propTypes = {
+    className: PropTypes.string,
+    notificationsGetRequest: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+    projectJoinApproveRequest: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+    projectJoinRejectRequest: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+};
+
+const defaultProps = {
+    className: '',
+    notificationsGetRequest: {},
+    projectJoinApproveRequest: {},
+    projectJoinRejectRequest: {},
+};
+
+const requests = {
+    notificationsGetRequest: {
+        url: '/notifications/',
+        method: requestMethods.GET,
+        onMount: true,
+        isUnique: true,
+        onSuccess: ({ props: { onRequestSuccess } }) => {
+            if (onRequestSuccess) {
+                onRequestSuccess();
+            }
+        },
+    },
+};
+
+const notificationItems = {
+    project_join_request: ProjectJoinRequestItem,
+    project_join_response: ProjectJoinResponseItem,
+    project_join_request_abort: ProjectJoinRequestAbortItem,
+};
+
+const NotificationItem = ({ notification }) => {
+    const Item = notificationItems[notification.notificationType];
+
+    if (Item) {
+        return <Item notification={notification} />;
+    }
+
+    return null;
+};
+
+NotificationItem.propTypes = {
+    notification: PropTypes.shape({
+        notificationTypes: PropTypes.string,
+    }).isRequired,
+};
+
+
+const NotificationEmpty = () => (
+    <div className={styles.emptyComponent} >
+        {_ts('notifications', 'noNotificationsText')}
+    </div>
+);
+
+const notificationKeySelector = n => n.id;
+const notificationItemRendererParams = (_, d) => ({ notification: d });
+
+const emptyObject = {};
+const emptyList = [];
+
+const requestsToListen = [
+    'projectJoinApproveRequest',
+    'projectJoinRejectRequest',
+    'notificationsGetRequest',
+];
+
+@RequestCoordinator
+@RequestClient(requests, requestsToListen)
+export default class Notifications extends React.PureComponent {
+    static propTypes = propTypes;
+    static defaultProps = defaultProps;
+
+    componentWillReceiveProps(nextProps) {
+        // TODO: use request's onPropChange, once the feature gets implemented
+        const {
+            projectJoinApproveRequest: newProjectJoinApproveRequest,
+            projectJoinRejectRequest: newProjectJoinRejectRequest,
+            notificationsGetRequest,
+        } = nextProps;
+
+        const {
+            projectJoinApproveRequest: oldProjectJoinApproveRequest,
+            projectJoinRejectRequest: oldProjectJoinRejectRequest,
+        } = this.props;
+
+        if (newProjectJoinApproveRequest.pending !== oldProjectJoinApproveRequest.pending
```

Comment:
According to the discussion with the "NOTIFICATION IMPLEMENTATION TEAM" at 14:44 we came to the conclusion that we do not fetch all the notifications an action is carried out.